src/modelinstance.cpp

```diff
@@ -1461,41 +1461,51 @@ Status ModelInstance::inferAsync(const RequestType* requestProto,
     OVMS_InferenceRequestCompletionCallback_t userCallback = requestProto->getResponseCompleteCallback();
     void* userCallbackData = requestProto->getResponseCompleteCallbackData();
     // here pass by copy into callback
-    // TODO unload model guard & test
-    inferRequest.set_callback([this, requestProto, &inferRequest, userCallback, userCallbackData](std::exception_ptr exception) {
-        SPDLOG_INFO("Entry of ov::InferRequest callback call");
-        if (exception) {
+    {
+        inferRequest.set_callback([this, requestProto, &inferRequest, userCallback, userCallbackData, modelUnloadGuardPtrMoved = std::shared_ptr<ModelInstanceUnloadGuard>(std::move(modelUnloadGuardPtr))](std::exception_ptr exception) mutable {
+            SPDLOG_DEBUG("Entry of ov::InferRequest callback call");
+            if (exception) {
+                try {
+                    SPDLOG_DEBUG("rethrow_exception");
+                    std::rethrow_exception(exception);
+                } catch (const std::exception& e) {
+                    SPDLOG_DEBUG("got exception in ov::InferRequest callback: {}", e.what());
+                } catch (...) {
+                    SPDLOG_DEBUG("got exception in ov::InferRequest callback");
+                    return;
+                }
```

Comment:
Is it possible to just simply use `exception` object as is? Why do we need to re-throw?

Reply:
I think this was used as BKM example when using callback in OV.